indra/llui/lllayoutstack.cpp

```diff
@@ -834,7 +836,7 @@ void LLLayoutStack::updatePanelRect( LLLayoutPanel* resized_panel, const LLRect&
 	LLLayoutPanel* other_resize_panel = NULL;
 	LLLayoutPanel* following_panel = NULL;
 
-	BOOST_REVERSE_FOREACH(LLLayoutPanel* panelp, mPanels)
+	BOOST_REVERSE_FOREACH(LLLayoutPanel* panelp, mPanels) // Should replace this when C++20 reverse view adaptor becomes available...
```

Comment:
It's tempting to respond to your comment by coding in terms of `rbegin()` and `rend()`, but I see your point: once we have the range library, it will be easier to replace `BOOST_REVERSE_FOREACH()` than the equivalent reverse-iterator logic.

Reply:
What are the plans for switching to C++20? I think there are some issues that would need to be fixed when I tried to use C++20.

Reply:
Sorry.
Why is this process done in reverse?
Looking at the processing details, wouldn't it work even if it's not a reverse loop?

Reply:
> It's tempting to respond to your comment by coding in terms of `rbegin()` and `rend()`, but I see your point: once we have the range library, it will be easier to replace `BOOST_REVERSE_FOREACH()` than the equivalent reverse-iterator logic.

Sorry.
Why is this process done in reverse?
Looking at the processing details, wouldn't it work even if it's not a reverse loop?

Reply:
> What are the plans for switching to C++20?

I want to work on that, but transitioning away from Jira requires a whole lot of attention just now.

Reply:
> Why is this process done in reverse?
Looking at the processing details, wouldn't it work even if it's not a reverse loop?

I haven't studied the semantics in detail, but it seems to me that the assignment to `following_panel`, and its assignment to `other_resize_panel`, would be different with a forwards traversal.